python/cudf/cudf/dataframe/categorical.py

```diff
@@ -92,20 +113,26 @@ def __init__(self, **kwargs):
         ordered : bool
             whether the categorical has a logical ordering (e.g. less than)
         """
-        categories = kwargs.pop("categories")
-        ordered = kwargs.pop("ordered")
-        dtype = pd.core.dtypes.dtypes.CategoricalDtype(categories, ordered)
-        kwargs.update({"dtype": dtype})
+        from cudf.dataframe.string import StringColumn
+
+        ordered = bool(kwargs.pop('ordered'))
+        categories = kwargs.pop('categories', [])
+        # Default to String dtype if len(categories) == 0, like pandas does
+        categories = columnops.as_column(categories) if len(categories) > 0 \
+            else StringColumn(data=nvstrings.to_device([]), null_count=0)
```

Comment:
Instead of needing to explicitly use `nvstrings` here, I believe you can pass a dtype to `columnops.as_column(...)` and have it create it as a StringColumn that way if the length is 0.

Reply:
Updated to use `columnops.column_empty()` :+1: